pkg/operation/botanist/extensions/operatingsystemconfig/executor/templates/scripts/execute-cloud-config.tpl.sh

```diff
@@ -104,7 +104,7 @@ if ! diff "$PATH_CLOUDCONFIG" "$PATH_CLOUDCONFIG_OLD" >/dev/null || ! diff "$PAT
 {{- end }}
     echo "Successfully restarted all units referenced in the cloud config."
     cp "$PATH_CLOUDCONFIG" "$PATH_CLOUDCONFIG_OLD"
-    cp "$PATH_CCD_SCRIPT_CHECKSUM" "$PATH_CCD_SCRIPT_CHECKSUM_OLD"
+    md5sum ${PATH_CCD_SCRIPT} > "$PATH_CCD_SCRIPT_CHECKSUM_OLD"
```

Comment:
Shouldn't it already be enough to move the line up before the daemons are restarted? This change is somehow hard to understand and brittle for maintenance. WDYT?

Reply:
I prefer to keep the update of the `*_OLD` files together.
Also, after the restarts it is guaranteed that all required changes are applied without failures.

Reply:
Okay, can you then put a small comment inline why copying is not enough like done a line above?

Reply:
I was thinking the same, but also considered that this file is downloaded every 30 seconds, thus I decided to not add comment, but seems it is required.